docs/debugging.md

```diff
@@ -271,4 +271,90 @@ slower than a compiled version, so you should default to installing the wheel in
 CI instead of compiling Cython, which can take up to a few minutes on some CI
 runners.
 
+### Compiling CPython and foundational packages with thread sanitizer (TSAN)
+
+[Thread sanitizer](https://github.com/google/sanitizers/wiki/ThreadSanitizerCppManual) (or TSAN) helps
+to detect C/C++ data races in concurrent systems. This tool can help to reveal free-threading
+related bugs in CPython and foundational packages (e.g. `numpy`).
+In this section we provide the commands to build free-threading compatible CPython and packages with TSAN and other hints to discover potential data races.
+
+#### Compile free-threading CPython with TSAN
+
+- Clone free-threading stable branch, e.g. 3.13.1
+
+```bash
+git clone https://github.com/python/cpython.git -b v3.13.1
```

Comment:
I think the branch to use here is `3.13`, rather than the latest tag (which may be behind with things already fixed): https://github.com/python/cpython/tree/3.13

Also above then "3.13.1" -> "3.13"